kernel/build-kernel.sh

```diff
@@ -0,0 +1,252 @@
+#!
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+#---------------------------------------------------------------------
+# Description: This script is the *ONLY* to build a kernel for development.
+# Note: The default kernel version will be taked from:
+# https://github.com/kata-containers/runtime/blob/master/versions.yaml
+#---------------------------------------------------------------------
+
+# Display usage to stdout.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly script_name="$(basename "${BASH_SOURCE[0]}")"
+readonly script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+#flags and default values
+# GOPATH used by Go tools.
+GOPATH=${GOPATH:-${HOME}/go}
+# Directory where the kernel source is located
+kernel_path=""
+# Kernel version to be used
+kernel_version=""
+# Flag know if need to download the kernel source
+download_kernel=false
+# The repository where kernel configuration lives
+runtime_repository="github.com/kata-containers/runtime"
+# The repository where kernel configuration lives
+kernel_config_repo="github.com/kata-containers/packaging"
+# Defult path to search patches to apply to kernel
+readonly default_patches_dir="${GOPATH}/src/${kernel_config_repo}/kernel/patches/"
+# Defult path to search config for kata
+readonly default_kernel_config_dir="${GOPATH}/src/${kernel_config_repo}/kernel/configs/"
+# 
+patches_path=""
+#
+hypervisor_target=""
+#
+arch_target=""
+#
+kernel_config_path=""
+# destdir
+DESTDIR="${DESTDIR:-/}"
+#PREFIX=
+PREFIX="${PREFIX:-/usr}"
+#project_name
+readonly project_name="kata-containers"
```

Comment:
Idea: you could move this higher up the script and then use `$project_name` in `$runtime_repository` and `$kernel_config_repo`.